doc/devel/coding_guide.rst

```diff
@@ -1,20 +1,76 @@
-.. _reviewers-guide:
+.. _pr-guidelines:
 
-********************
-Reviewers guideline
-********************
+***********************
+Pull request guidelines
+***********************
 
-.. _pull-request-checklist:
+Pull requests (PRs) are the mechanism for contributing to Matplotlibs code and
+documentation.
 
-Pull request checklist
-======================
+.. _pr-checklist-authors:
 
-Branch selection
-----------------
+Checklist for PR authors
+========================
+
+.. note::
+
+   * We value contributions from people with all levels of experience. In
+     particular if this is your first PR not everything has to be perfect.
+     We'll guide you through the PR process.
+   * Nevertheless, try to follow the below guidelines as well as you can to
+     help make the PR process quick and smooth.
+   * Be patient with reviewers. We try our best to respond quickly, but we
+     have limited bandwidth. If there is no feedback within a couple of days,
+     please ping us by posting a comment to your PR.
+
+When making a PR, pay attention to:
+
+* :ref:`Target the master branch <pr-branch-selection>`.
+* Adhere to the :ref:`coding_guidelines`.
+* Update the :ref:`documentation <pr-documentation>` if necessary.
+* Aim at making the PR as "ready-to-go" as you can. This helps to speed up
+  the review process.
+* It is ok to open incomplete or work-in-progress PRs if you need help or
+  feedback from the developers. You may mark these as
+  `draft pull requests <https://help.github.com/en/articles/about-pull-requests#draft-pull-requests>`_
+  on GitHub.
+
+See also :ref:`contributing` for how to make a PR.
+
+.. _pr-checklist-reviewers:
+
+Checklist for PR reviewers
+==========================
+.. note::
+
+   * If you have commit rights, then you are trusted to use them.
+     **Please help review and merge PRs!**
+   * Be patient and `kind <https://youtu.be/tzFWz5fiVKU?t=49m30s>`__ with
+     contributors.
 
-Generally, all pull requests (*PRs*) should target the master branch. Other
-branches are fed through automatic or manual backports. Directly targeting
-other branches is only rarely necessary for special maintenance work.
+Content topics:
+
```

Comment:
Does rest have a checkbox formatting the way markdown does? I'd make this explicit checkboxes to guide the eye

Reply:
Good idea! Unfortunately, ReST does not have native checkbox support (and also only some flavors of Markdown have).

There are however ways to patch that in. I would like to defer that discussion to a separate PR because there are different ways, each with some advantages / disadvantages, but all with some kind of overhead. We should decide if we want to add that.

I would like to keep this PR focused on the structure, not on layout.